### PR TITLE
Deleting a VM also deletes its snapshots

### PR DESCRIPTION
**Pull request by marsaud**

### CODE REVIEW

File: src/xapi.js [hunk 1]
```diff
@@ -592,6 +592,12 @@ export default class Xapi extends XapiBase {
       }))
     }
 
+    await Promise.all(map(vm.$snapshots, snapshot => {
+      try {
+        return this.deleteVm(snapshot.uuid, true).catch(err => {console.log('err', err);noop})
+      } catch (_) {console.log('_', _.message)}
+    }))
+
```
Comment (julien-f) [inline src/xapi.js]: I wonder if it should be optional via a boolean like `deleteSnapshots`?

Comment (marsaud) [inline src/xapi.js]: I wondered too, but according to @olivierlambert , a orphan snapshot is useless...

File: src/xapi.js [hunk 2]
```diff
@@ -592,6 +592,12 @@ export default class Xapi extends XapiBase {
       }))
     }
 
+    await Promise.all(map(vm.$snapshots, snapshot => {
+      try {
```
Comment (julien-f) [inline src/xapi.js]: `deleteVm()` is asynchronous, it cannot throw synchronously an exception.

File: src/xapi.js [hunk 3]
```diff
@@ -592,6 +592,12 @@ export default class Xapi extends XapiBase {
...
+        return this.deleteVm(snapshot.uuid, true).catch(err => {console.log('err', err);noop})
```
Comment (julien-f) [inline src/xapi.js]: Use `$id` instead of `uuid`.

Comment (julien-f) [inline src/xapi.js]: `noop` is useless here.

**Pull request merged**